src/scorecard.js

```diff
@@ -0,0 +1,90 @@
+class Scorecard {
+  constructor() {
+    this._pinsKnocked = []
+    this._frame = 0
+    this._roll = 0
+    this._frameScores = []
+    this._strikesSpares = {}
+    this._maxFrames = 9
+  };
+
+  inputRoll(number) {
+    if(this._frameNotExist) {
```

Comment:
You are not calling the method `this._frameNotExist` here, so this condition is in fact testing whether that property is defined, which it is, so it's always true, which means you are always pushing []... Weird 🤔 

Anyway - a good one for getting visibility.